ykrt/src/mt.rs

```diff
@@ -286,16 +288,28 @@ impl MT {
                     unreachable!();
                 }
                 HotLocation::Tracing(opt) => {
-                    match mtt.tracing.get() {
-                        Some(other_hl_ptr) => {
+                    // FIXME: This is the sort of hack that keeps me awake at night: we really want
+                    // to return from the outer function, and to modify `hl`, but can't because
+                    // we're in a closure. The integer return value allows us to perform the
+                    // control flow we want.
+                    let r = THREAD_MTTHREAD.with(|mtt| {
```

Comment:
I've been up against this sort of thing myself before. It's a pain.

Does the code touching the thread local have to be a closure, or can it be a regular function? I wonder if it would be better pulled-out of here into a `fn is_current_thread_tracing()` function?

Also, I understand that the scope of those magic numbers is small, but it's really hard to read. Can we define enum so that we get the opportunity to give the cases more easily understandable names?

Something like:

```
enum OngoingThreadTracing{
   NotTracing,
   AnotherLocation,
   ThisLocation,
}
```

What do you think?

Reply:
This is a temporary hack and I'm not sure an `enum` will make it much less gross to be honest. Turning the closure into a function won't help, unfortunately, because you still can't return from an outer function.

Reply:
Can we at the very least put a comment next to where you consume each magic number saying what case it covers?

Reply:
Fixed in 4fd2f67.